examples/stm32/f4/stm32f429i-discovery/ts-i2c/gfx.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * This is the core graphics library for all our displays, providing a common
+ * set of graphics primitives (points, lines, circles, etc.).  It needs to be
+ * paired with a hardware-specific library for each display device we carry
+ * (to handle the lower-level functions).
+ *
+ * Adafruit invests time and resources providing this open source code, please
+ * support Adafruit & open-source hardware by purchasing products from Adafruit!
+ *
+ * Copyright (c) 2013 Adafruit Industries.  All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * - Redistributions of source code must retain the above copyright notice,
+ *   this list of conditions and the following disclaimer.
+ * - Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Modified the AdaFruit library to be a C library, changed the font and
+ * generally munged it in a variety of ways, creating a reasonably quick
+ * and dirty way to put something "interesting" on the LCD display.
+ * --Chuck McManis (2013, 2014)
+ *
+ */
+
+#include <stdint.h>
+#include <math.h>
+#include <stdlib.h>
+#include "gfx.h"
+
+#define pgm_read_byte(addr) (*(const unsigned char *)(addr))
```

Comment:
oh you meant just that one line.. i thought you meant gfx.c itself.. ignore me! ;)